packages/vaadin-themable-mixin/test/themable-mixin.test.js

```diff
@@ -266,101 +228,117 @@ describe('ThemableMixin', () => {
   });
 
   it('material should override vaadin module styles', () => {
-    expect(getComputedStyle(components[6].$.text).opacity).to.equal('0.5');
+    expect(getComputedStyle(getText(components['test-style-override'])).opacity).to.equal('0.5');
   });
 
   it('should override lumo module styles', () => {
-    expect(getComputedStyle(components[6].$.text).color).to.equal('rgb(0, 0, 0)');
+    expect(getComputedStyle(getText(components['test-style-override'])).color).to.equal('rgb(0, 0, 0)');
   });
 
   it('should override material module styles', () => {
-    expect(getComputedStyle(components[6].$.text).color).to.equal('rgb(0, 0, 0)');
+    expect(getComputedStyle(getText(components['test-style-override'])).color).to.equal('rgb(0, 0, 0)');
   });
 
   it('should respect vaadin style module order', () => {
-    expect(getComputedStyle(components[6]).display).to.equal('block');
+    expect(getComputedStyle(components['test-style-override']).display).to.equal('block');
   });
 
   it('should respect custom style module order', () => {
-    expect(getComputedStyle(components[6]).position).to.equal('relative');
+    expect(getComputedStyle(components['test-style-override']).position).to.equal('relative');
   });
 
   it('should not include duplicate styles', () => {
     const name = 'test-duplicate';
 
     // Create a wildcard style that applies to both the parent and the child
-    const duplicateStyle = css`
-      :host {
-        color: blue;
-      }
-    `;
-    createStyles('vaadin-test-duplicate-styles', 'test-duplicate*', duplicateStyle);
+    const duplicateStyle = ':host { color: blue; }';
+    createStyles('vaadin-test-duplicate-styles', 'test-duplicate*', unsafeCSS(duplicateStyle));
 
     // Define the test-duplicate-parent and test-duplicate components
     defineCustomElement(name + '-parent');
-    defineCustomElement(name, name + '-parent');
+    defineCustomElement(name, name + '-parent', '<div></div>');
 
     // Create an instance of the test-duplicate component
     const testComponent = fixtureSync(`<${name}></${name}>`);
-    // Get all the styles from the component as one big string (let's assume it may have multiple style tags)
-    const styles = [...testComponent.shadowRoot.querySelectorAll('style')].map((style) => style.textContent).join('');
+
+    // Get all the style rules from the component
+    // Gather from the <style> tags (PolymerElement) and from the adoptedStyleSheets (LitElement)
+    const rules = [
+      ...(testComponent.shadowRoot.adoptedStyleSheets || []),
+      ...[...testComponent.shadowRoot.querySelectorAll('style')].map((style) => style.sheet)
+    ]
+      .map((sheet) => [...sheet.cssRules])
+      .flat();
+
     // Check the number of occurences of the style rule
-    const occurrences = styles.split(duplicateStyle.toString()).length - 1;
+    const occurrences = rules.reduce((acc, rule) => acc + (rule.cssText === duplicateStyle ? 1 : 0), 0);
```

Comment:
Using `filter(...).length` might be easier to understand?

Reply:
Done.